pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -498,3 +502,89 @@ func (r *MigCluster) UpdateProvider(provider pvdr.Provider) {
 		}
 	}
 }
+
+func (m *MigCluster) GetInternalRegistryPath(c k8sclient.Client) (string, error) {
+	client, err := m.GetClient(c)
+	if err != nil {
+		return "", err
+	}
+	isList := imgapi.ImageStreamList{}
+	err = client.List(
+		context.TODO(),
+		k8sclient.InNamespace("openshift"),
+		&isList)
+	if err == nil && len(isList.Items) > 0 {
```

Comment:
I wonder if this would give undesired results? We want the logic to be something closer to the following right?

1. get the imagestreamlist
2. if err != nil return "", err because you don't deterministically know why the error occurred
3 if len(isList.Items > 0) have the parsing logic

Reply:
So I *think* this logic is the way we want it as-is -- @dymurray I think you wrote this originally for the plugin, so you might be able to comment further here.

Basically if we can determine the value from an existing imagestream, then that's how we determine the internal registry path. If we can't do this, we fall back to the ConfigMap for OCP4+ (although we discussed changing that in the future to a different lookup), and the Cluster IP for OCP3. 

So the question is what happens if the is LIst call errors out. Currently that also goes to the same fallback we use when there aren't any imagestreams in the openshift namespace. I think that's probably the better choice, since it gives us the same sort of result we'd get if there weren't any imagestreams. I don't know that erroring out instead improves anything, since we have a fallback here, but I don't have a strong opinion here. Lets see what @dymurray thinks.

Reply:
@sseago @alaypatel07 I agree with Scott's comments that I prefer falling back rather than erroring out. I understand the perspective that we'd want to bubble the error back since we don't know what caused it but if we have a fallback mechanism for errors or an empty list I think we'd want to attempt that before erroring out completely.